rc/base/fish.kak

```diff
@@ -34,29 +34,42 @@ addhl -group /fish/code regex \b(and|begin|bg|bind|block|break|breakpoint|builti
 # ‾‾‾‾‾‾‾‾
 
 def -hidden _fish_filter_around_selections %{
-    # remove trailing white spaces
-    try %{ exec -draft -itersel <a-x> s \h+$ <ret> d }
+    eval -draft -itersel %{
+        exec <a-x>
+        # remove trailing white spaces
+        try %{ exec -draft s \h + $ <ret> d }
+    }
 }
 
 def -hidden _fish_indent_on_char %{
     eval -draft -itersel %{
-        # deindent on (else|end) command insertion
-        try %{ exec -draft <space> <a-i>w <a-k> (else|end) <ret> <a-lt> }
+        # align middle and end structures to start and indent when necessary
+        try %{ exec -draft <a-x> <a-k> ^ \h * (else) $ <ret> <a-\;> <a-?> ^ \h * (if)                                 <ret> s \A | \Z <ret> \' <a-&>                   }
+        try %{ exec -draft <a-x> <a-k> ^ \h * (end)  $ <ret> <a-\;> <a-?> ^ \h * (begin|for|function|if|switch|while) <ret> s \A | \Z <ret> \' <a-&>                   }
+        try %{ exec -draft <a-x> <a-k> ^ \h * (case) $ <ret> <a-\;> <a-?> ^ \h * (switch)                             <ret> s \A | \Z <ret> \' <a-&> \' <space> <a-gt> }
     }
 }
 
 def -hidden _fish_indent_on_new_line %{
     eval -draft -itersel %{
         # preserve previous line indent
-        try %{ exec -draft <space> K <a-&> }
+        try %{ exec -draft K <a-&> }
```

Comment:
why remove the `<space>` here ? that wont work well if you are appending on a multiline selection